tests/validation/tests/rke/common.py

```diff
@@ -146,6 +147,143 @@ def wait_for_etcd_cluster_health(node, etcd_private_ip=False):
     return result
 
 
+def verify_metrics_server_addon_images(k8sversion, rke_client, kubectl,
+                                       namespace, selector):
+
+    metricserver = get_component_version(rke_client, k8sversion,
+                                         "rancher/metrics-server")
+    verify_resource_with_kubectl(kubectl, namespace, selector, metricserver)
+
+
+def verify_ingress_addon_images(k8sversion, rke_client, kubectl, namespace,
+                                selector1, selector2):
+
+    ingressdefaultbackend = \
+        get_component_version(rke_client, k8sversion,
+                              "rancher/nginx-ingress-controller-defaultbackend")
+    nginxingresscontoller =\
+        get_component_version(rke_client, k8sversion,
+                              "rancher/nginx-ingress-controller")
+
+    verify_resource_with_kubectl(kubectl, namespace, selector1,
+                                 nginxingresscontoller)
+    verify_resource_with_kubectl(kubectl, namespace, selector2,
+                                 ingressdefaultbackend)
+
+
+def verify_dns_addon_images(k8sversion, rke_client, kubectl, namespace,
+                            selector):
+
+    coredns = get_component_version(rke_client, k8sversion,
+                                    "rancher/coredns-coredns")
+    verify_resource_with_kubectl(kubectl, namespace, selector, coredns)
+
+
+def verify_networking_addon_images(k8sversion, rke_client, kubectl,
+                                   namespace, selector):
+
+    flannel = get_component_version(rke_client, k8sversion,
+                                    "rancher/coreos-flannel")
+
+    calico = get_component_version(rke_client, k8sversion,
+                                   "rancher/calico-node")
+    verify_resource_with_kubectl(kubectl, namespace, selector, calico, flannel)
+
+
+def verify_resource_with_kubectl(kubectl, namespace, selector, resource1,
```

Comment:
This name should be updated to better reflect what it's doing, also pass a list of resources instead of just 1 or 2.

Reply:
Renamed to verify_component_status_with_kubectl